core/src/main/java/cz/xtf/core/openshift/ClusterVersionInfo.java

```diff
@@ -69,15 +70,32 @@ boolean isMajorMinorOnly() {
      * @return true if version is in major.minor.micro format, false if not valid url
      */
     boolean isMajorMinorMicro() {
-        if (openshiftVersion != null) {
-            return versionMatcher.group(2) != null && versionMatcher.group(3) == null;
+        if (openshiftVersion != null && versionMatcher.groupCount() >= 2) {
+            return versionMatcher.group(2) != null;
         }
         return false;
     }
 
-    boolean isNightly() {
-        if (openshiftVersion != null) {
-            return versionMatcher.group(3) != null;
+    /**
+     * Check if version is release candidate. Version in x.y.z-rc... format
+     *
+     * @return true if version contains -rc. string
+     */
+    boolean isReleaseCandidate() {
+        if (openshiftVersion != null && versionMatcher.groupCount() >= 3) {
+            return Optional.ofNullable(versionMatcher.group(3)).orElse("").contains("-rc.");
+        }
+        return false;
+    }
+
+    /**
+     * Check if version is release candidate. Version in x.y.z-rc... format
+     *
+     * @return true if version contains -rc. string
+     */
```

Comment:
```suggestion
    /**
     * @return true if the version should be searched in dev-preview
     */
```

Reply:
well, `ClusterVersionInfo` doesn't really know how this method will be used. That it will be used to determine client URL. It could be used on more places, despite its package private scope.

Reply:
understandable, anyway it could use some fix `@return true if version contains -rc. string` is not correct

Reply:
should be fixed now